qiskit_experiments/characterization/t1_experiment.py

```diff
@@ -0,0 +1,161 @@
+from qiskit_experiments import AnalysisResult
+
+
+class T1Analysis(BaseAnalysis):
+    """T1 Experiment result analysis class."""
+
+    def _run_analysis(
+        self, experiment_data, t1_guess=None, amplitude_guess=None, offset_guess=None, **kwargs
+    ) -> Tuple[AnalysisResult, None]:
+        """
+        Calculate T1
+
+        Args:
+            experiment_data (ExperimentData): the experiment data to analyze
+            t1_guess: Optional, an initial guess of T1
+            amplitude_guess: Optional, an initial guess of the coefficient of the exponent
+            offset_guess: Optional, an initial guess of the offset
+
+        Returns:
+            The analysis result with the estimated T1
+        """
+
+        size = len(experiment_data._data)
+        delays = np.zeros(size, dtype=float)
+        means = np.zeros(size, dtype=float)
+        stddevs = np.zeros(size, dtype=float)
+
+        for i, circ in enumerate(experiment_data._data):
+            delays[i] = circ["metadata"]["delay"]
+            count0 = circ["counts"].get("0", 0)
+            count1 = circ["counts"].get("1", 0)
+            shots = count0 + count1
+            means[i] = count1 / shots
+            stddevs[i] = np.sqrt(means[i] * (1 - means[i]) / shots)
+            # problem for the fitter if one of the std points is
+            # exactly zero
+            if stddevs[i] == 0:
+                stddevs[i] = 1e-4
+
+        if t1_guess is None:
+            t1_guess = np.mean(delays)
+        if offset_guess is None:
+            offset_guess = means[-1]
+        if amplitude_guess is None:
+            amplitude_guess = means[0] - offset_guess
+
+        fit_out, fit_err, chisq = BaseAnalysis.curve_fit_wrapper(
+            BaseAnalysis.exp_fit_fun,
+            delays,
+            means,
+            stddevs,
+            p0=[amplitude_guess, t1_guess, offset_guess],
+        )
+
+        analysis_result = AnalysisResult(
+            {
+                "amplitude": fit_out[0],
+                "t1": fit_out[1],
+                "offset": fit_out[2],
+                "amplitude_err": fit_err[0],
+                "t1_err": fit_err[1],
+                "offset_err": fit_err[2],
+                "chisq": chisq,
+                "unit": experiment_data._data[0]["metadata"]["unit"]
+            }
+        )
+
+        analysis_result["is_good_fit"] = (
+            abs(analysis_result["amplitude"] - 1.0) < 0.1
+            and abs(analysis_result["offset"]) < 0.1
+            and analysis_result["chisq"] < 3
+            and (analysis_result["amplitude_err"] is None or analysis_result["amplitude_err"] < 0.1)
+            and (analysis_result["offset_err"] is None or analysis_result["offset_err"] < 0.1)
+            and (
+                analysis_result["t1_err"] is None
+                or analysis_result["t1_err"] < analysis_result["t1"]
+            )
+        )
+
+        return analysis_result, None
```

Comment:
It would be good to generate a exp plot of the fitted data (eg see #18 for example, though it has some issues with current ExperimentData class).

Reply:
I began to do it, and decided to wait for #19.